3 New OpenCV Effects (Tracker, Stabilization, and Object Detection) (#585)
OpenShot/libopenshot: tests/CMakeLists.txt

	endif()
endif()

> ####################### OPENCV ##########################

Reviewer: Same suggestion on all CmakeLists.txt files... let's rebase this file on 'develop', and then add back in these changes.... so we aren't reverting any updates.

  KeyFrame_Tests.cpp
  Point_Tests.cpp
  Settings_Tests.cpp
  Timeline_Tests.cpp)

> ##########  SET OPENCV RELATED TEST FILES  ###############
> set(OPENSHOT_CV_TEST_FILES
>   CVTracker_Tests.cpp
>   CVStabilizer_Tests.cpp
>   # CVObjectDetection_Tests.cpp
>   )

Reviewer 2: ```suggestion
##########  SET OPENCV RELATED TEST FILES  ###############
if(OpenCV_FOUND)
  set(OPENSHOT_CV_TEST_FILES
    CVTracker_Tests.cpp
    CVStabilizer_Tests.cpp
    # CVObjectDetection_Tests.cpp
    )
  set(OPENSHOT_CV_LIBRARIES
    ${OpenCV_LIBS}
    ${PROTOBUF_LIBRARY}
  )
endif()
```
Then you can just use those variables unconditionally in the executable definition. CMake will expand any undefined variable to an empty string.

################ TESTER EXECUTABLE #################
# Create unit test executable (openshot-test)
message (STATUS "Tests enabled, test executable will be built as tests/openshot-test")

> if (OpenCV_FOUND)
>   add_executable(openshot-test
>     tests.cpp
>     ${OPENSHOT_TEST_FILES} 
>     ${OPENSHOT_CV_TEST_FILES})

Reviewer 2: ```suggestion
add_executable(openshot-test
  tests.cpp
  ${OPENSHOT_TEST_FILES} 
  ${OPENSHOT_CV_TEST_FILES}
  )
```
With above.

>   # Link libraries to the new executable
>   target_link_libraries(openshot-test openshot ${UnitTest++_LIBRARIES} ${OpenCV_LIBS} ${PROTOBUF_LIBRARY})

Reviewer 2: ```suggestion
# Link libraries to the new executable
target_link_libraries(openshot-test
  openshot
  ${UnitTest++_LIBRARIES}
  ${OPENSHOT_CV_LIBRARIES}
  )
```
Again, unconditionally. Really, I'll be wanting to get you set up with targets for all of this stuff (for one thing, that'll make it exportable), but for now this will serve.

Reviewer 2: The other option is to do the configuration parts in multiple stages, like so:
```cmake
add_executable(openshot-test
  tests.cpp
  ${OPENSHOT_TEST_FILES}
  )
target_link_libraries(openshot-test openshot)

if(OpenCV_FOUND)
  target_sources(openshot-test
    CVTracker_Tests.cpp
    CVStabilizer_Tests.cpp
    # CVObjectDetection_Tests.cpp
    )
  target_link_libraries(openshot-test PUBLIC
    ${OpenCV_LIBS}
    ${PROTOBUF_LIBRARY}
    )
endif()
```
That also works perfectly fine, in fact it's slightly cleaner IMHO.

else()
  add_executable(openshot-test
    tests.cpp
    ${OPENSHOT_TEST_FILES} )

  # Link libraries to the new executable
  target_link_libraries(openshot-test openshot ${UnitTest++_LIBRARIES})

> endif()

Reviewer 2: And then all of this can be dropped.
